File: src-tauri/src/domain/response/transaction.rs
use crate::model::transaction::{
    TransactionMonthAmountResponse, TransactionMonthMethodResponse, TransactionResponse,
    TransactionResponseDeleteAt, TransactionResponseMonthStatusFailed,
    TransactionResponseMonthStatusSuccess, TransactionResponseYearStatusFailed,
    TransactionResponseYearStatusSuccess, TransactionYearMethodResponse,
    TransactionYearlyAmountResponse,
};

use super::PaginationMeta;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransactionMonthStatusSuccess {
    pub status: String,
    pub message: String,
    pub data: Vec<TransactionResponseMonthStatusSuccess>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransactionYearStatusSuccess {
    pub status: String,
    pub message: String,
    pub data: Vec<TransactionResponseYearStatusSuccess>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransactionMonthStatusFailed {
    pub status: String,
    pub message: String,
    pub data: Vec<TransactionResponseMonthStatusFailed>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransactionYearStatusFailed {
    pub status: String,
    pub message: String,
    pub data: Vec<TransactionResponseYearStatusFailed>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransactionMonthMethod {
    pub status: String,
    pub message: String,
    pub data: Vec<TransactionMonthMethodResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransactionYearMethod {
    pub status: String,
    pub message: String,
    pub data: Vec<TransactionYearMethodResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransactionMonthAmount {
    pub status: String,
    pub message: String,
    pub data: Vec<TransactionMonthAmountResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransactionYearAmount {
    pub status: String,
    pub message: String,
    pub data: Vec<TransactionYearlyAmountResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransaction {
    pub status: String,
    pub message: String,
    pub data: Option<TransactionResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransactions {
    pub status: String,
    pub message: String,
    pub data: Vec<TransactionResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransactionDelete {
    pub status: String,
    pub message: String,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseTransactionAll {
    pub status: String,
    pub message: String,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponsePaginationTransaction {
    pub status: String,
    pub message: String,
    pub data: Vec<TransactionResponse>,
    pub pagination: Option<PaginationMeta>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponsePaginationTransactionDeleteAt {
    pub status: String,
    pub message: String,
    pub data: Vec<TransactionResponseDeleteAt>,
    pub pagination: Option<PaginationMeta>,
}

File: src/types/domain/response/transaction/transaction.ts
import {
  Transaction,
  TransactionMonthAmount,
  TransactionMonthMethod,
  TransactionMonthStatusFailed,
  TransactionMonthStatusSuccess,
  TransactionTrashed,
  TransactionYearlyAmount,
  TransactionYearMethod,
  TransactionYearStatusFailed,
  TransactionYearStatusSuccess,
} from "@/types/model";
import { PaginationMeta } from "../pagination";

export interface ApiResponseTransactionMonthStatusSuccess {
  status: string;
  message: string;
  data: TransactionMonthStatusSuccess[];
}

export interface ApiResponseTransactionYearStatusSuccess {
  status: string;
  message: string;
  data: TransactionYearStatusSuccess[];
}

export interface ApiResponseTransactionMonthStatusFailed {
  status: string;
  message: string;
  data: TransactionMonthStatusFailed[];
}

export interface ApiResponseTransactionYearStatusFailed {
  status: string;
  message: string;
  data: TransactionYearStatusFailed[];
}

export interface ApiResponseTransactionMonthMethod {
  status: string;
  message: string;
  data: TransactionMonthMethod[];
}

export interface ApiResponseTransactionYearMethod {
  status: string;
  message: string;
  data: TransactionYearMethod[];
}

export interface ApiResponseTransactionMonthAmount {
  status: string;
  message: string;
  data: TransactionMonthAmount[];
}

export interface ApiResponseTransactionYearAmount {
  status: string;
  message: string;
  data: TransactionYearlyAmount[];
}

export interface ApiResponseTransaction {
  status: string;
  message: string;
  data?: Transaction;
}

export interface ApiResponseTransactions {
  status: string;
  message: string;
  data: Transaction[];
}

export interface ApiResponseTransactionDelete {
  status: string;
  message: string;
}

export interface ApiResponseTransactionAll {
  status: string;
  message: string;
}

export interface ApiResponsePaginationTransaction {
  status: string;
  message: string;
  data: Transaction[];
  pagination: PaginationMeta;
}

export interface ApiResponsePaginationTransactionDeleteAt {
  status: string;
  message: string;
  data: TransactionTrashed[];
  pagination: PaginationMeta;
}
